homeassistant/components/almond/__init__.py

```diff
@@ -201,9 +202,39 @@ def __init__(
 class AlmondAgent(conversation.AbstractConversationAgent):
     """Almond conversation agent."""
 
-    def __init__(self, api: WebAlmondAPI):
+    def __init__(self, hass: core.HomeAssistant, api: WebAlmondAPI, entry):
         """Initialize the agent."""
+        self.hass = hass
         self.api = api
+        self.entry = entry
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return {"name": "Powered by Almond", "url": "https://almond.stanford.edu/"}
+
+    async def async_get_onboarding(self):
+        """Get onboard url if not onboarded."""
+        if self.entry.data.get("onboarded"):
+            return None
+
+        host = self.entry.data["host"]
+        if self.entry.data.get("is_hassio"):
+            host = "/core_almond"
+        elif self.entry.data["type"] != TYPE_LOCAL:
+            host = f"{host}/me"
+        return {
+            "text": "Send anonymized commands to the authors of Almond for training.",
+            "url": f"{host}/conversation",
+        }
+
+    async def async_set_onboarding(self, new_data):
```

Comment:
Can we be more specific data wise ? Now the caller decides the key to store onboarding result, and we define the key in this class when we check for onboarding. Instead we should just pass a boolean ?  Or even just mark_onboarding_shown() ?